src/lib/y2storage/disk_analyzer.rb

```diff
@@ -223,120 +173,97 @@ def mbr_gap
     # @see #scope
     #
     # @return [Hash{String => Array<Partition>}] see {#partitions_with_id}
-    def windows_partitions
-      @windows_partitions ||= begin
-        result = find_windows_partitions
-        log.info("Found Windows partitions: #{result}")
-        result
-      end
+    def windows_partitions(*disks, limit: DEFAULT_DISKS_LIMIT)
+      data_for(*disks, :windows_partitions, limit: limit) { |d| find_windows_partitions(d) }
     end
 
-  private
+    # Disks that are suitable for installing Linux.
+    #
+    # @return [Array<Storage::Disk>] candidate disks
+    def candidate_disks(limit: DEFAULT_DISKS_LIMIT)
+      return @candidate_disks if @candidate_disks
```

Comment:
Yes, you are right. In this case what would be the best solution? Avoid a limit parameter and always use a value internally?